src/main/java/com/artozersky/HackerNewsAPI/service/impl/PostServiceImpl.java

```diff
@@ -26,58 +34,128 @@ public class PostServiceImpl implements NewsPostService {
     @Autowired
     private ModelMapper modelMapper;
 
+    // @Autowired
+    private SimpleCache cacheService;
```

Comment:
What you need to do is to wrap the cache implementation you made to be as a service to simplify the usage and abstract the cache to the user (you this time), service layer is business layer, cache is business service so it needs an abstraction as a separate service with interface and impl in the service folder and use it here in the service layer.

Reply:
fixed